src/model/channel/message.rs

```diff
@@ -72,6 +72,18 @@ pub struct Message {
     /// Array of [`Role`]s' Ids mentioned in the message.
     pub mention_roles: Vec<RoleId>,
     /// Channels specifically mentioned in this message.
+    ///
+    /// **Note**:
+    /// Not all channel mentions in a message will appear in `mention_channels`.
+    /// Only textual channels that are visible to everyone in a lurkable guild will ever be included.
+    /// A lurkable guild being one that allows users to read public channels in a server without
+    /// actually joining the server. It also allows users to look at these channels without being
+    /// logged in to Discord.
+    /// Only crossposted messages (via Channel Following) currently include `mention_channels` at all.
+    /// If no mentions in the message meet these requirements, this field will not be sent.
+    /// [`See more in the discord documentation`].
+    ///
+    /// [`See more in the discord documentation`]: https://discord.com/developers/docs/resources/channel#message-object
```

Comment:
Content is fine, but the paragraphs can be divided and the text wrapped at around 100 columns better. Also, to allow for changing the text inside the link, I suggest using reference-text. Additionally, to be consistent with the rest of the documentation that recommends users to read other sources of documentation or literature, the link text should use the "Refer to xxx" language.
```suggestion
    ```suggestion
    /// **Note**:
    /// Not all channel mentions in a message will appear in `mention_channels`. Only textual 
    /// channels that are visible to everyone in a lurkable guild will ever be included.
    /// 
    /// A lurkable guild is one that allows users to read public channels in a server without
    /// actually joining the server. It also allows users to look at these channels without being
    /// logged in to Discord.
    ///
    /// Only crossposted messages (via Channel Following) currently include `mention_channels` at 
    /// all. If no mentions in the message meet these requirements, this field will not be sent.
    /// [Refer to Discord's documentation for more information][discord-docs].
    ///
    /// [discord-docs]: https://discord.com/developers/docs/resources/channel#message-object
```
```

Reply:
Ah this looks way neater, will commit it